engine/runtime/src/main/java/org/enso/interpreter/EnsoLanguage.java

```diff
@@ -117,8 +118,8 @@ public final class EnsoLanguage extends TruffleLanguage<EnsoContext> {
   private static final LanguageReference<EnsoLanguage> REFERENCE =
       LanguageReference.create(EnsoLanguage.class);
 
-  private final ContextThreadLocal<ExecutionEnvironment[]> executionEnvironment =
-      locals.createContextThreadLocal((ctx, thread) -> new ExecutionEnvironment[1]);
+  private final ContextLocal<ExecutionEnvironment[]> executionEnvironment =
+      locals.createContextLocal(ctx -> new ExecutionEnvironment[1]);
```

Comment:
- The 197bd53244948dc3541bd3ace3fa9c7cc75604bf change is probably correct
   - when the _guest code pool_ is larger than one
   - the code can be executed on any of the threads
   - and all of them want to share this info
- I am however not sure about other consequences of this change
   - probably the consequences aren't that huge as #12340 hasn't yet been implemented...

Reply:
This is a problem right now with a guest pool equal to 1.